### ISSUE TITLE
"Expected a 'mps:0' generator device but found 'cpu'" using shuffle=True on DataLoader (Pytorch 2.11)

### ISSUE THREAD

**Issue by D-Vaillant** [pytorch/pytorch#115287]
### 🐛 Describe the bug

This is the exact same bug as https://github.com/pytorch/pytorch/issues/110820, except I'm on the latest version of Pytorch that's supposedly fixed it.

### Versions

PyTorch version: 2.1.1
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: macOS 14.0 (arm64)
GCC version: Could not collect
Clang version: 14.0.3 (clang-1403.0.22.14.1)
CMake version: version 3.27.8
Libc version: N/A

Python version: 3.10.13 (main, Aug 24 2023, 12:59:26) [Clang 15.0.0 (clang-1500.0.40.1)] (64-bit runtime)
Python platform: macOS-14.0-arm64-arm-64bit
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Apple M2

Versions of relevant libraries:
[pip3] numpy==1.26.2
[pip3] torch==2.1.1
[conda] Could not collect

**Comment by jbschlosser** [comment]
Hey @D-Vaillant, thanks for the report! I'm going to close this one and reopen the other issue; could you please continue the conversation there?
